projects/graph/graph.py

```diff
@@ -63,23 +63,56 @@ def dft_recursive(self, starting_vertex):
         beginning from starting_vertex.
         This should be done using recursion.
         """
-        pass  # TODO
+        def go(starting_vertex, acc):
+            if starting_vertex in acc:
+                return
+            print(starting_vertex)
+            for neighbor in self.vertices[starting_vertex]:
+                acc.add(starting_vertex)
+                go(neighbor, acc)
+        go(starting_vertex, set())
 
```

Comment:
very nice; but couldnt you forgo the nested method and just call the initial method on itself?